### PR TITLE
movie api endpoints

### LINKED ISSUE #25: Movie Api endpoint
# I want to be able to interact with db for movies data

## Acceptance criteria:
Validation is applied.
Tested on postman.
Queries are working fine.

### PR DESCRIPTION
**Pull request by AfaqAhmed-XGrid**
## **Issue** #25 
     Title: Api endpoints related to movies data

## Description
This pull request sets up all required api endpoint to interact with db for movies data:
=> Api Endpoints: Created api endpoints for getting movies data, get graph data and saving new data
=> Validation: Added validation for all routes using express validator
=> Queries: Handled queries like search, sort and filter.

## **Checklist**
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the README.md, or the docs directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes (if applicable)
- [x] Every function, interface, class has a comment describing what it does and input/output parameters

### CODE REVIEW

File: api/modules/movie/movie.helper.js [hunk 1]
```diff
@@ -59,4 +75,27 @@ const constructSorting = (req) => {
...
+
+  if (!minYear || !maxYear) {
+    return { $match: {} };
+  }
+
+  return {
+    $match: {
+      releaseDate: {
+        $gte: new Date(minYear, 0, 1), // Start of startYear
+        $lte: new Date(maxYear, 11, 31), // End of endYear
```
Comment (waqar-360) [inline api/modules/movie/movie.helper.js:95]: why you are hardcoding the dates and month, you can use moment to get startOf and endOf month and years

File: api/modules/movie/movie.route.js [hunk 2]
```diff
@@ -1,13 +1,54 @@
...
+const movieMiddleware = require('./movie.middleware');
+const commonMiddleware = require('../../common/commonMiddleware');
 
 // Creating isntance
 const router = express.Router();
 
 // Api endpoint to get movies data
-router.get('/getallmovies', getAllMovies);
+router.get(
+    '/get-all-movies',
```
Comment (waqar-360) [inline api/modules/movie/movie.route.js]: change get-all-movies ==> all-movies . Get is already assumed by the GET call

File: api/modules/movie/movie.route.js [hunk 3]
```diff
@@ -1,13 +1,54 @@
...
+    movieMiddleware.validateGetAllMovies,
+    commonMiddleware.validationError,
+    getAllMovies,
+);
+
+// Api endpoint to get graph data
+router.get(
```
Comment (waqar-360) [inline api/modules/movie/movie.route.js:39]: same

File: api/modules/movie/movie.route.js [hunk 4]
```diff
@@ -1,13 +1,54 @@
...
+    '/get-graph-data',
+    movieMiddleware.validateGetGraphData,
+    commonMiddleware.validationError,
+    getGraphData,
+);
+
+// Api endpoint to post movie data
+router.post(
```
Comment (waqar-360) [inline api/modules/movie/movie.route.js:47]: post-movie-data ==> movie-data

**Review by waqar-360 — CHANGES_REQUESTED** [review]

File: api/modules/movie/movie.route.js (hunk 4 above)
Comment (AfaqAhmed-XGrid) [reply api/modules/movie/movie.route.js:47]: Done

File: api/modules/movie/movie.route.js (hunk 3 above)
Comment (AfaqAhmed-XGrid) [reply api/modules/movie/movie.route.js:39]: Done

File: api/modules/movie/movie.route.js (hunk 2 above)
Comment (AfaqAhmed-XGrid) [reply api/modules/movie/movie.route.js]: Done

**Review by junaidtariqxgrid — APPROVED** [review]

**Review by junaidtariqxgrid — APPROVED** [review]

**Review by waqar-360 — COMMENTED** [review]
LGTM

**Review by waqar-360 — APPROVED** [review]

**Pull request merged**